frontend/server/src/Controllers/User.php

```diff
@@ -3958,6 +3958,28 @@ public static function isMainIdentity(
      * @omegaup-request-param int|null $page
      */
     public static function getRankForTypeScript(\OmegaUp\Request $r) {
+        try {
+            $r->ensureIdentity();
+        } catch (\OmegaUp\Exceptions\UnauthorizedException $e) {
+            // Do nothing. Not logged user can access here
+            $r->identity = null;
+        }
+        if (is_null($r->identity)) {
+            return self::getRankToLoggedOutUserForTypeScript($r);
+        }
+        return self::getRankToLoggedInUserForTypeScript($r);
+    }
+
+    /**
+     * when the user is logged out, prepare all the properties to be sent to the rank table view via TypeScript
+     *
+     * @return array{templateProperties: array{payload: UserRankTablePayload, title: \OmegaUp\TranslationString}, entrypoint: string}
+     *
+     * @omegaup-request-param null|string $filter
+     * @omegaup-request-param int|null $length
+     * @omegaup-request-param int|null $page
+     */
+    public static function getRankToLoggedOutUserForTypeScript(\OmegaUp\Request $r) {
```

Comment:
Las reglas que utilizamos es la siguiente: 

Las funciones que se mandan llamar directamente desde un end point llevan el sufijo `ForTypeScript`. 

Al ser estas funciones auxiliares, no es necesario que lleven el sufijo, así que con esto es suficiente:
```suggestion
    public static function getRankToLoggedOutUser(\OmegaUp\Request $r) {
```

Otra regla sería que sólo las funciones con el sufijo `ForTypeScript` reciben como parámetro un `request`. Entonces en estas nuevas funciones tendrás que enviar los parámteros explícitos que vayas a ocupar, en este caso serían los que te sugerí en el comentario anterior.

Esto último ayuda para que PSALM (que es el validador de tipos) esté mas contento